drivers/painter/ili9xxx/qp_ili9486.c

```diff
@@ -0,0 +1,161 @@
+    const uint8_t madctl[] = {
+        [QP_ROTATION_0]   = ILI9XXX_MADCTL_BGR,
+        [QP_ROTATION_90]  = ILI9XXX_MADCTL_BGR | ILI9XXX_MADCTL_MV,
+        [QP_ROTATION_180] = ILI9XXX_MADCTL_BGR,
+        [QP_ROTATION_270] = ILI9XXX_MADCTL_BGR | ILI9XXX_MADCTL_MV,
+    };
+    const uint8_t functl[] = {
+        [QP_ROTATION_0]   = 0x42,
+        [QP_ROTATION_90]  = 0x62,
+        [QP_ROTATION_180] = 0x22,
+        [QP_ROTATION_270] = 0x02,
+    };
+
+    // clang-format off
+    uint8_t rotation_sequence[] = {
+        // Command,                 Delay,  N, Data[N]
+        ILI9XXX_SET_MEM_ACS_CTL,        0,  1, madctl[rotation],
+        ILI9XXX_SET_FUNCTION_CTL,       0,  2, 0x00, functl[rotation],
+        ILI9XXX_CMD_SLEEP_OFF,          5,  0,
+        ILI9XXX_CMD_DISPLAY_ON,         5,  0,
+    };
+    // clang-format on
+    qp_comms_bulk_command_sequence(device, rotation_sequence, sizeof(rotation_sequence));
+
+    return true;
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Driver vtable
+
+const struct tft_panel_dc_reset_painter_driver_vtable_t ili9486_driver_vtable = {
+    .base =
+        {
+            .init            = qp_ili9486_init,
+            .power           = qp_tft_panel_power,
+            .clear           = qp_tft_panel_clear,
+            .flush           = qp_tft_panel_flush,
+            .pixdata         = qp_tft_panel_pixdata,
+            .viewport        = qp_tft_panel_viewport,
+            .palette_convert = qp_tft_panel_palette_convert_rgb565_swapped,
+            .append_pixels   = qp_tft_panel_append_pixels_rgb565,
+        },
+    .num_window_bytes   = 2,
+    .swap_window_coords = false,
+    .opcodes =
+        {
+            .display_on         = ILI9XXX_CMD_DISPLAY_ON,
+            .display_off        = ILI9XXX_CMD_DISPLAY_OFF,
+            .set_column_address = ILI9XXX_SET_COL_ADDR,
+            .set_row_address    = ILI9XXX_SET_PAGE_ADDR,
+            .enable_writes      = ILI9XXX_SET_MEM,
+        },
+};
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// SPI
+
+#ifdef QUANTUM_PAINTER_ILI9486_SPI_ENABLE
+
+// Factory function for creating a handle to the ILI9486 device
+painter_device_t qp_ili9486_make_spi_device(uint16_t panel_width, uint16_t panel_height, pin_t chip_select_pin, pin_t dc_pin, pin_t reset_pin, uint16_t spi_divisor, int spi_mode) {
+    for (uint32_t i = 0; i < ILI9486_NUM_DEVICES; ++i) {
+        tft_panel_dc_reset_painter_device_t *driver = &ili9486_drivers[i];
+        if (!driver->base.driver_vtable) {
+            driver->base.driver_vtable         = (const struct painter_driver_vtable_t *)&ili9486_driver_vtable;
+            driver->base.comms_vtable          = (const struct painter_comms_vtable_t *)&spi_comms_with_dc_vtable;
+            driver->base.native_bits_per_pixel = 16; // RGB565
```

Comment:
Im pretty sure the code from @tzarc's i used as a base to add shift_reg support was tested by him. No idea what his hardware is, tho. The code for the "default" screens was not changed, but just added stuff on top

Also thought about making a separate files for the driver. But thought it would be cleaner to have everything on a single file as the screen itself is the same model.

By the way, any ideas on why the 90/270 rotations won't work (note: i haven't tested further as i've been busy today + spent free time on cleaning this mess of a code)